internal/locate/region_request.go

```diff
@@ -2462,3 +2465,59 @@ func (s *replicaSelector) recordAttemptedTime(duration time.Duration) {
 		proxyReplica.attemptedTime += duration
 	}
 }
+
+// canFastRetry returns true if the request can be sent to next replica.
+func (s *replicaSelector) canFastRetry() bool {
+	accessLeader, ok := s.state.(*accessKnownLeader)
```

Comment:
Should we just do fast retry when `tikv_client_read_timeout` is configured? It looks a little bit dangerous to use
```
if not fast retry
   return false
return true
```
as most of the request would be fast retried .

Besides, maybe we could consult slow information to decide whether fast retry should work. /cc @zyguan @MyonKeminta

Reply:
I see the original logic of [canFallback2Follower](https://github.com/tikv/client-go/pull/923/files#diff-0c36d07fbd1fc71c850ff57d323fe12bdf7f2c053ef71b7fec5bb44358f1070dR1152-R1154) is trying to do fast backoff when retry next replica, so fast backoff is not only work for `tikv_client_read_timeout` is configured.
cc @you06

Reply:
> I see the original logic of [canFallback2Follower](https://github.com/tikv/client-go/pull/923/files#diff-0c36d07fbd1fc71c850ff57d323fe12bdf7f2c053ef71b7fec5bb44358f1070dR1152-R1154) is trying to do fast backoff when retry next replica, so fast backoff is not only work for `tikv_client_read_timeout` is configured. cc @you06

@crazycs520 It the current logic the same? If so we could continue

Reply:
Not same, this PR is for the fix issue #1166.